crates/ruff_python_formatter/src/expression/string/docstring.rs

```diff
@@ -855,6 +1241,58 @@ fn indentation_length(line: &str) -> TextSize {
     TextSize::new(indentation)
 }
 
+/// Trims at most `indent_len` indentation from the beginning of `line`.
+///
+/// This treats indentation in precisely the same way as `indentation_length`.
+/// As such, it is expected that `indent_len` is computed from
+/// `indentation_length`. This is useful when one needs to trim some minimum
+/// level of indentation from a code snippet collected from a docstring before
+/// attempting to reformat it.
+fn indentation_trim(indent_len: TextSize, mut line: &str) -> &str {
+    let mut seen_indent_len = 0u32;
+    for char in line.chars() {
+        if seen_indent_len >= indent_len.to_u32() {
+            return line;
+        }
+        if char == '\t' {
+            // Pad to the next multiple of tab_width
+            seen_indent_len += 8 - (seen_indent_len.rem_euclid(8));
+            line = &line[1..];
+        } else if char.is_whitespace() {
```

Comment:
Python only supports `\t` and ` ` (space) as valid indentation characters (and `\f` form feeds that reset the indentation). It's unclear to me if supporting all white space is okay because we're inside a docstring or should limit it to Python-whitespace only (the same applies for `indent_with_suffix`, we have `trim_whitespace_start` helper to do so)